test/relayserver/ServerTestEnvironments.ts

```diff
@@ -0,0 +1,233 @@
+  });
+
+  const relayHub = RelayHub__factory.connect(relayHubAddress, provider);
+
+  await relayHub
+    .connect(relayOwner)
+    .stakeForAddress(relayManagerAddress, unstakeDelay, {
+      value: utils.parseEther('1'),
+    });
+
+  return relayServer;
+};
+
+const newServerInstanceNoFunding = async (
+  template: IForwarder,
+  serverWorkdirs?: ServerWorkdirs
+) => {
+  const { workdir } = getTemporaryWorkdirs();
+
+  const managerKeyManager = createKeyManager(serverWorkdirs?.managerWorkdir);
+  const workersKeyManager = createKeyManager(serverWorkdirs?.workersWorkdir);
+  const txStoreManager = new TxStoreManager({
+    workdir: serverWorkdirs?.workdir ?? workdir,
+  });
+
+  const dependencies: ServerDependencies = {
+    txStoreManager,
+    managerKeyManager,
+    workersKeyManager,
+  };
+
+  const factory = await createSmartWalletFactory(template);
+  const relayHub = await deployRelayHub(undefined, {});
+  const deployVerifierFactory = await hardhat.getContractFactory(
+    'DeployVerifier'
+  );
+  const deployVerifier = await deployVerifierFactory.deploy(factory.address);
+  const relayVerifierFactory = await hardhat.getContractFactory(
+    'RelayVerifier'
+  );
+  const relayVerifier = await relayVerifierFactory.deploy(factory.address);
+
+  const originalConfig = { ...config };
+
+  config.util.extendDeep(originalConfig, {
+    contracts: {
+      relayHubAddress: relayHub.address,
+      deployVerifierAddress: deployVerifier.address,
+      relayVerifierAddress: relayVerifier.address,
+      smartWalletFactoryAddress: factory.address,
+    },
+    app: {
+      devMode: true,
+      checkInterval: 10,
+      logLevel: 5,
+    },
+  });
+
+  const relayServer = new RelayServer(dependencies);
+
+  return relayServer;
+};
+
+const createKeyManager = (workdir?: string): KeyManager => {
+  if (workdir != null) {
+    return new KeyManager(1, workdir);
+  } else {
+    return new KeyManager(1, undefined, utils.randomBytes(32).toString());
+  }
+};
+
+const createRelayHttpRequest = async (
+  envelopingRequest: UserDefinedEnvelopingRequest,
+  relayClient: RelayClient,
+  hubInfo: HubInfo
+): Promise<EnvelopingTxRequest> => {
+  type RelayClientExposed = {
+    _prepareHttpRequest: (
+      hubInfo: HubInfo,
+      envelopingRequest: EnvelopingRequest
+    ) => Promise<EnvelopingTxRequest>;
+    _getEnvelopingRequestDetails: (
+      envelopingRequest: UserDefinedEnvelopingRequest
+    ) => Promise<EnvelopingRequest>;
+  };
+
+  const localClient = relayClient as unknown as RelayClientExposed;
+
+  const envelopingRequestDetails =
+    await localClient._getEnvelopingRequestDetails(envelopingRequest);
+
+  return await localClient._prepareHttpRequest(
+    hubInfo,
+    envelopingRequestDetails
+  );
+};
+
+const relayTransaction = async (
+  envelopingRequest: UserDefinedEnvelopingRequest,
+  relayServer: RelayServer,
+  relayClient: RelayClient,
+  assertRelayed = true
+) => {
+  const hubInfo = relayServer.pingHandler();
```

Comment:
should we change the name of ´pingHandler´ to follow the RelayClient one?

Reply:
Yes, we should do it for consistency

Reply:
Should we include this change in the [server PR](https://github.com/rsksmart/rif-relay-server/pull/75)

Reply:
yes Ill include it, sorry I was waiting for Jurajs PR, but Ill do it today, and then rebase Jurajs PR.

Reply:
updated